src/invert/laplace/impls/hypre3d/hypre3d_laplace.hxx

```diff
@@ -204,6 +206,33 @@ public:
   bout::HypreVector<Field3D> rhs;
   bout::HypreSystem<Field3D> linearSystem;
 
+  // used to save some statistics
+  int n_solves = 0;
+  int cumulative_iterations = 0;
+  BoutReal average_iterations = 0.0;
+  class Hypre3dMonitor : public Monitor {
+  public:
+    Hypre3dMonitor(LaplaceHypre3d &laplace_in) : laplace(laplace_in) {}
+
+    int call(Solver*, BoutReal, int, int) override {
+      if (laplace.n_solves == 0) {
+        laplace.average_iterations = 0.0;
+        return 0;
+      }
+
+      // Calculate average and reset counters
+      laplace.average_iterations = static_cast<BoutReal>(laplace.cumulative_iterations)
+                                   / static_cast<BoutReal>(laplace.n_solves);
+
+      output_info << endl << "Hypre3d average iterations: " << laplace.average_iterations << endl;
+
+      return 0;
```

Comment:
I'd prefer to stick the body of this in the .cxx file

I might also suggest using `output_info.write("\nHypre3d average iterations: {}\n", laplace.average_iterations)` as being a bit more readable :)